File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/C++/s596918302_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int dpx(vector<vector<int>>& arr, int n) {
    vector<vector<int>> dp(n, vector<int>(3));

    dp[0][0] = arr[0][0];
    dp[0][1] = arr[0][1];
    dp[0][2] = arr[0][2];

    for (int i = 1; i < n; i++) {
        for (int in = 0; in < 3; in++) {
            int j = 0, k = 0;
            if (in == 1) {
                j = 0;
                k = 2;
            } else if (in == 0) {
                j = 1;
                k = 2;
            } else {
                j = 0;
                k = 1;
            }
            dp[i][in] = max(dp[i - 1][k], dp[i - 1][j]) + arr[i][in];
        }
    }

    return max(dp[n - 1][0], max(dp[n - 1][1], dp[n - 1][2]));
}

int main() {
    int n;
    cin >> n;
    vector<vector<int>> arr(n, vector<int>(3));

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < 3; j++) {
            cin >> arr[i][j];
        }
    }

    int ans = dpx(arr, n);
    cout << ans << endl;

    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Python/s596918302_commented.py
import sys

def main():
    n = int(sys.stdin.readline())
    arr = []
    for _ in range(n):
        a, b, c = map(int, sys.stdin.readline().split())
        arr.append([a, b, c])
    
    ans = dpx(arr, n)
    print(ans)

def dpx(arr, n):
    dp = [[0]*3 for _ in range(n)]
    dp[0][0] = arr[0][0]
    dp[0][1] = arr[0][1]
    dp[0][2] = arr[0][2]
    
    for i in range(1, n):
        for in_col in range(3):
            if in_col == 0:
                j, k = 1, 2
            elif in_col == 1:
                j, k = 0, 2
            else:
                j, k = 0, 1
            current_max = max(dp[i-1][j], dp[i-1][k])
            dp[i][in_col] = current_max + arr[i][in_col]
    
    return max(dp[-1])

if __name__ == "__main__":
    main()
# End of Code
